frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx

```diff
@@ -221,9 +254,13 @@ export function EntityListLoaderRtkQuery<Entity, EntityWrapper>({
   }, [dispatch, rtkError, requestStatePath, queryKey]);
 
   useEffect(() => {
-    if (data) {
+    if (data && !isFetching) {
```

Comment:
This was a bug - `data` was being set for a different `requestStatePath`. It got caught by tests.

I did the same change in `EntityObjectLoader` in this PR for safety.